lib/portlayer/attach/attach.go

```diff
@@ -70,6 +70,12 @@ func toggle(handle *exec.Handle, connected bool) (*exec.Handle, error) {
 	b := serial.GetVirtualDevice().Backing.(*types.VirtualSerialPortURIBackingInfo)
 
 	serviceURI := fmt.Sprintf("tcp://%s:%d", ip, constants.SerialOverLANPort)
+	proxyURI := fmt.Sprintf("telnet://%s:%d", ip, constants.SerialOverLANPort)
+
+	if b.ProxyURI == proxyURI && c.Connected == connected {
+		log.Debugf("Already in the desired state (connected: %t, proxyURI: %s)", connected, proxyURI)
+		return handle, nil
+	}
 
 	if b.ServiceURI == serviceURI && c.Connected == connected {
```

Comment:
In fact, the serviceURI is forwarded to the proxy to tell what the address of the remote system should be. In our case, it is known and fixed (localhost:2379) which is the address of the telnet server